framework/log/Dispatcher.php

```diff
@@ -123,12 +124,15 @@ public function getLogger()
      * Sets the connected logger.
      * @param Logger|string|array $value the logger to be used. This can either be a logger instance
      * or a configuration that will be used to create one using [[Yii::createObject()]].
+     * @throws InvalidConfigException
      */
     public function setLogger($value)
     {
         if (is_string($value) || is_array($value)) {
             $value = Yii::createObject($value);
+            Yii::setLogger($value);
```

Comment:
I think this should be moved below so already configured object can also be set at the main logger.

Reply:
Agree. Right after `$this->_logger = $value;`

Reply:
Ok. Thanks